src/RIGID/fix_shake.cpp

```diff
@@ -121,49 +121,40 @@ FixShake::FixShake(LAMMPS *lmp, int narg, char **arg) :
   mass_list = new double[atom->ntypes];
   nmass = 0;
 
-  char mode = '\0';
   int next = 6;
   while (next < narg) {
-    if (strcmp(arg[next],"b") == 0) mode = 'b';
-    else if (strcmp(arg[next],"a") == 0) mode = 'a';
-    else if (strcmp(arg[next],"t") == 0) mode = 't';
-    else if (strcmp(arg[next],"m") == 0) {
-      mode = 'm';
-      atom->check_mass(FLERR);
-
-    // break if keyword that is not b,a,t,m
-
-    } else if (isalpha(arg[next][0])) break;
-
-    // read numeric args of b,a,t,m
-
-    else if (mode == 'b') {
-      int i = utils::inumeric(FLERR,arg[next],false,lmp);
+    if ((strcmp(arg[next],"b") == 0) && (next+1 < narg)) {
+      int i = utils::expand_type_int(FLERR,arg[next+1],1,lmp);
```

Comment:
I have been using enum values (e.g., Atom::ATOM) instead of numbers for the mode argument of expand_type_int

Reply:
Thanks for the reminder!
Laziness at work. Of course this should be using the enumerator. Will make the change ASAP.